Automattic/woocommerce-payments
# Run post-upgrade actions for all requests

Fixes #2153.

#### Changes proposed in this Pull Request

While working on #1991, I noticed that the post-upgrade hook `woocommerce_woocommerce_payments_updated` only runs when visiting a wp-admin page.

Because the migrations in #1991 have to run before any front-end or wp-admin request, this PR introduces a change to the upgrade behavior, running them on `init` rather than on `admin_init`.

It is also better to run this migration during a non-admin request because if the plugin is upgraded e.g. using WP-CLI, the migrations wouldn't run.

Kudos to @ricardo and @DanReyLop for the insight and great discussion on how to tackle this in p1623435917484000-slack-CGGCLBN58 🙌









#### Testing instructions





As currently the only module (other than the new code implemented in #1991) using `woocommerce_woocommerce_payments_updated` is the Apple Pay domain association, let's use it for testing to make sure there are no regressions.

* Set up your site so you can use Apple Pay ([testing using live credentials is a requirement](https://docs.woocommerce.com/document/payments/apple-pay/#troubleshooting)).
* Make sure you enable payment request buttons.

**To verify previous behavior:**

1. Check out an earlier commit prior to this PR.
2. A part of the Apple Pay verification procedure is a file being created on the server. Remove the file `/.well-known/apple-developer-merchantid-domain-association` from your test site.
3. Manually change the plugin version in the database to an earlier version, e.g. `2.4.0`:
    `UPDATE wp_options SET option_value = '2.4.0' WHERE option_name = 'woocommerce_woocommerce_payments_version'`
4. Visit a front-end page. The `/.well-known/apple-developer-merchantid-domain-association` file should still be missing.
5. Visit any wp-admin page. The above file should now be present.

**To verify new behavior:**

Do the same as above, but in the "Visit a front-end page" step, the file should be there.

-------------------
